File: Android/iEatAndroid/src/main/java/com/thoughtworks/ieat/utils/PropertyUtils.scala
package com.thoughtworks.ieat.activity.utils

import java.util.Properties
import android.R
import android.content.res.Resources

object PropertyUtils{
  var properties : Properties = null

  def init(mResources: Resources) {
    if (properties == null) {
      properties = new Properties()
      val appConfigProperties = mResources.openRawResource(com.thoughtworks.ieat.R.raw.app_config)
      try {
        properties.load(appConfigProperties)
      } catch {
        case e : IndexOutOfBoundsException => throw new RuntimeException(e)
      }
    }
  }

  def getServerHost() : String = {
    properties.getProperty("server.host")
  }

  def getServerPort() : Int =  {
    Integer.valueOf(properties.getProperty("server.port"))
  }
}

File: Android/iEatAndroid/src/main/java/com/thoughtworks/ieat/utils/PropertyUtils.java
package com.thoughtworks.ieat.utils;

import android.content.res.Resources;
import com.thoughtworks.ieat.R;

import java.io.IOException;
import java.io.InputStream;
import java.util.Properties;

public class PropertyUtils {
    private static Properties properties = null;

    public static void init(Resources mResources) {
        if (properties == null) {
            properties = new Properties();
            InputStream appConfigProperties = mResources.openRawResource(R.raw.app_config);
            try {
                properties.load(appConfigProperties);
            } catch (IndexOutOfBoundsException e) {
                throw new RuntimeException(e);
            } catch (IOException e) {
                throw new RuntimeException(e);
            }
        }
    }

    public static String getServerHost() {
        return properties.getProperty("server.host");
    }

    public static int getServerPort() {
        return Integer.valueOf(properties.getProperty("server.port"));
    }
}